validation-test/Sema/type_checker_crashers_fixed/issue-73454.swift

```diff
@@ -0,0 +1,21 @@
+// RUN: %target-typecheck-verify-swift
+
+// https://github.com/apple/swift/issues/73454
+
+func object_method_bind_ptrcall_v(
+  _: UnsafeMutableRawPointer?...
+) {
+}
+
+func setDensity(_ density: Double, ptr: UnsafeMutableRawPointer) {
+  var copy_density = density
+  var arr: [Int] = []
+
+  object_method_bind_ptrcall_v(&copy_density)
+  object_method_bind_ptrcall_v(&arr)
+
+  object_method_bind_ptrcall_v(&arr, &copy_density)
```

Comment:
Does this actually generate safe code though? :) CC @atrick

Reply:
It should be no less safe then the rest of cases we have here, I am happy to add SILGen tests too.

Reply:
This happens to be correct because the compiler has a special case that calls the intrinsic `_convertMutableArrayToPointerArgument`

It's super misleading because the syntax is dangerously incorrect for any other indirectly referenced storage like
`&string` or `&data`.

But... for the purpose of this PR it looks fine.